### PR TITLE
spell check docs with spellintian (SC-770)

### PR DESCRIPTION
**Pull request by holmanb**
```
Fix spelling errors

- Add Makefile target that checks for spelling errors
- Add Makefile target that fixes spelling errors
- Add spelling check to travis doc tests
- Fix various spelling errors in the docs 
```

## Test Steps
revert the bug fixes in the second commit to test the new makefile target functionality
```
git reset --hard HEAD^
make check_spelling
echo $?
make fix_spelling
make check_spelling # note the one error fix_spelling couldn't automagically fix
# manually fix doc/rtd/topics/datasources/smartos.rst
make fix_spelling
echo $?
```

Note that this [build](https://app.travis-ci.com/github/canonical/cloud-init/jobs/557796208) fails (before the spelling fixes) and this [build](https://app.travis-ci.com/github/canonical/cloud-init/jobs/557796695) succeeds (after the fixes).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -83,6 +83,24 @@ deb-src:
...
+			-e 'doc/examples/cloud-config.txt: Bye Bye'
+
+check_spelling:
+	@! make -s _check_spelling 2>/dev/null
+
+# Limitation: duplicate words with newline between them are not automatically fixed
+fix_spelling:
+	@make -s _check_spelling | \
+		sed 's/ (duplicate word)//g' | \
+		awk -F ': | -> ' '{printf "sed -i \047s/%s/%s/g\047 %s\n", $$2, $$3, $$1}' | \
```
Comment (TheRealFalcon) [inline Makefile:101]: Magic! Would comments help here, or do I just need to `man awk`?

**Review by TheRealFalcon — COMMENTED** [review]
There goes our possibility of ever joining Hacktoberfest...

I noticed on Travis:
```
0.46s$ make check_spelling
Makefile:88: recipe for target '_check_spelling' failed
The command "make check_spelling" exited with 0.
```

I don't think the check is running correctly there. Have we tried with a misspeled word to see if Travis fails on it?

**Comment by holmanb** [comment]
> There goes our possibility of ever joining Hacktoberfest...

We have plenty of text files with spelling errors outside of `doc` for the Hacktoberfesters to have a go at ;) 

> I noticed on Travis:
> 
> ```
> 0.46s$ make check_spelling
> Makefile:88: recipe for target '_check_spelling' failed
> The command "make check_spelling" exited with 0.
> ```
> 
> I don't think the check is running correctly there. Have we tried with a misspeled word to see if Travis fails on it?

I think this string is printed on older versions of `gawk`. In a bionic container I see:
```
root@me:~/cloud-init# git status
On branch holmanb/spellcheck
Your branch is up to date with 'upstream/holmanb/spellcheck'.

nothing to commit, working tree clean
root@me:~/cloud-init# make check_spelling
Makefile:88: recipe for target '_check_spelling' failed
root@me:~/cloud-init# git reset --hard HEAD^
HEAD is now at cd3fc0ad spell check docs with spellintian
root@me:~/cloud-init# make check_spelling
doc/rtd/conf.py: Supress -> Suppress
doc/rtd/topics/network-config-format-v1.rst: of of (duplicate word) -> of
doc/rtd/topics/datasources.rst: your your (duplicate word) -> your
doc/rtd/topics/faq.rst: occured -> occurred
doc/rtd/topics/instancedata.rst: filesytem -> filesystem
doc/rtd/topics/instancedata.rst: platfrom -> platform
doc/rtd/topics/datasources/vultr.rst: aquire -> acquire
doc/rtd/topics/datasources/smartos.rst: responds responds (duplicate word) -> responds
doc/rtd/topics/datasources/vmware.rst: then then (duplicate word) -> then
doc/examples/cloud-config-install-packages.txt: specifc -> specific
doc/examples/cloud-config.txt: ommitted -> omitted
doc/examples/cloud-config.txt: occurance -> occurrence
doc/examples/cloud-config.txt: filesytem -> filesystem
doc/examples/cloud-config.txt: persistant -> persistent
doc/examples/cloud-config.txt: acheived -> achieved
doc/examples/cloud-config-mount-points.txt: ommitted -> omitted
doc/examples/cloud-config-disk-setup.txt: overwriten -> overwritten
doc/examples/cloud-config-disk-setup.txt: formating -> formatting
doc/examples/seed/README: formated -> formatted
doc/examples/seed/meta-data: formated -> formatted
doc/examples/cloud-config-landscape.txt: formated -> formatted
doc/examples/cloud-config-power-state.txt: acheived -> achieved
doc/examples/cloud-config-user-groups.txt: specifed -> specified
doc/examples/kernel-cmdline.txt: everthing -> everything
doc/sources/ovf/README: pressing pressing (duplicate word) -> pressing
doc/sources/ovf/make-iso: an an (duplicate word) -> an
doc/sources/kernel-cmdline.txt: if if (duplicate word) -> if
Makefile:95: recipe for target 'check_spelling' failed
make: *** [check_spelling] Error 1
```
Reverting the spelling fixes shows that this still finds the errors, so the warning is a red herring. I'll come up with a "fix".

**Review by TheRealFalcon — APPROVED** [review]
Thanks! This will improve the quality of our docs, and hopefully make us look a little more professional (until people read my PR comments).

**Pull request merged**